modular_skyrat/modules/opposing_force/code/opposing_force_subsystem.dm

```diff
@@ -0,0 +1,196 @@
+		unsubmitted_applications += opposing_force
+
+	opposing_force.deny(denier, reason)
+
+	broadcast_queue_change(opposing_force)
+
+	return TRUE
+
+/datum/controller/subsystem/opposing_force/proc/modify_request(datum/opposing_force/opposing_force, changes)
+	if(LAZYFIND(submitted_applications, opposing_force))
+		submitted_applications -= opposing_force
+
+	if(LAZYFIND(approved_applications, opposing_force))
+		approved_applications -= opposing_force
+
+	if(!LAZYFIND(unsubmitted_applications, opposing_force))
+		unsubmitted_applications += opposing_force
+
+	broadcast_queue_change(opposing_force)
+
+/datum/controller/subsystem/opposing_force/proc/get_current_applications()
+	return LAZYLEN(submitted_applications) + LAZYLEN(approved_applications)
+
+/datum/controller/subsystem/opposing_force/proc/new_opfor(datum/opposing_force/opposing_force)
+	unsubmitted_applications += opposing_force
+
+/datum/controller/subsystem/opposing_force/proc/remove_opfor(datum/opposing_force/opposing_force)
+	if(LAZYFIND(unsubmitted_applications, opposing_force))
+		unsubmitted_applications -= opposing_force
+	if(LAZYFIND(submitted_applications, opposing_force))
+		submitted_applications -= opposing_force
+	if(LAZYFIND(approved_applications, opposing_force))
+		approved_applications -= opposing_force
+
+	broadcast_queue_change()
+
+/datum/controller/subsystem/opposing_force/proc/unsubmit_opfor(datum/opposing_force/opposing_force)
+	if(LAZYFIND(approved_applications, opposing_force))
+		approved_applications -= opposing_force
+	if(LAZYFIND(submitted_applications, opposing_force))
+		submitted_applications -= opposing_force
+	if(!LAZYFIND(unsubmitted_applications, opposing_force))
+		unsubmitted_applications += opposing_force
+
+
+	broadcast_queue_change()
+
+/datum/controller/subsystem/opposing_force/proc/view_opfor(datum/opposing_force/opposing_force, mob/viewer)
+	if(!is_admin(viewer.client))
+		message_admins("Oppoding_force_subsystem: [ADMIN_LOOKUPFLW(viewer)] attempted to view an OPFOR application but was not an admin!")
+		CRASH("Opposing_force_subsystem: Attempted to view an opposing force but the viewer was not an admin!")
+
+	opposing_force.ui_interact(viewer)
+
+/datum/controller/subsystem/opposing_force/proc/get_check_antag_listing()
+	var/list/returned_html = list("<br>")
+
+	returned_html += "<b>OPFOR Applications</b>"
+
+	returned_html += "Submitted - FOLLOW QUEUE!"
+	var/queue_count = 1
+	for(var/datum/opposing_force/opposing_force in submitted_applications)
+		returned_html += " - <b>[queue_count].</b> [opposing_force.build_html_panel_entry()]"
+		queue_count++
+
+	returned_html += "Approved"
+	for(var/datum/opposing_force/opposing_force in approved_applications)
+		returned_html += " - [opposing_force.build_html_panel_entry()]"
+
+	returned_html += "Unsubmitted"
+	for(var/datum/opposing_force/opposing_force in unsubmitted_applications)
+		returned_html += " - [opposing_force.build_html_panel_entry()]"
+
+	return returned_html.Join("<br>")
```

Comment:
This feels more like what we should use in this case, no? We use `\n` everywhere else, pretty much.
```suggestion
	return returned_html.Join("\n")
```